tests/tendermint/test_lib.py

```diff
@@ -122,3 +124,67 @@ def test_post_transaction_invalid_mode(b):
     tx = b.validate_transaction(tx)
     with pytest.raises(ValidationError):
         b.write_transaction(tx, 'nope')
+
+
+@pytest.mark.bdb
+def test_update_utxoset(tb, signed_create_tx, signed_transfer_tx, db_context):
+    mongo_client = MongoClient(host=db_context.host, port=db_context.port)
+    tb.update_utxoset(signed_create_tx)
+    utxoset = mongo_client[db_context.name]['utxos']
+    assert utxoset.count() == 1
+    utxo = utxoset.find_one()
+    assert utxo['transaction_id'] == signed_create_tx.id
+    assert utxo['output_index'] == 0
+    tb.update_utxoset(signed_transfer_tx)
+    assert utxoset.count() == 1
+    utxo = utxoset.find_one()
+    assert utxo['transaction_id'] == signed_transfer_tx.id
+    assert utxo['output_index'] == 0
+
+
+@pytest.mark.bdb
+def test_store_transaction(mocker, tb, signed_create_tx,
+                           signed_transfer_tx, db_context):
+    mocked_store_asset = mocker.patch('bigchaindb.backend.query.store_asset')
+    mocked_store_metadata = mocker.patch(
+        'bigchaindb.backend.query.store_metadatas')
+    mocked_store_transaction = mocker.patch(
+        'bigchaindb.backend.query.store_transaction')
+    mongo_client = MongoClient(host=db_context.host, port=db_context.port)
+    tb.store_transaction(signed_create_tx)
+    utxoset = mongo_client[db_context.name]['utxos']
+    assert utxoset.count() == 1
+    utxo = utxoset.find_one()
+    assert utxo['transaction_id'] == signed_create_tx.id
+    assert utxo['output_index'] == 0
+    mocked_store_asset.assert_called_once_with(
+        tb.connection,
+        {'id': signed_create_tx.id, 'data': signed_create_tx.asset['data']},
+    )
+    mocked_store_metadata.asser_called_once_with(
+        tb.connection,
+        {'id': signed_create_tx.id, 'metadata': signed_create_tx.metadata},
+    )
+    mocked_store_transaction.assert_called_once_with(
+        tb.connection,
+        {k: v for k, v in signed_create_tx.to_dict().items()
+         if k not in ('asset', 'metadata')},
+    )
+    mocked_store_asset.reset_mock()
+    mocked_store_metadata.reset_mock()
+    mocked_store_transaction.reset_mock()
+    tb.store_transaction(signed_transfer_tx)
```

Comment:
Same here? `store_bulk_transactions`

Reply:
A different test is actually needed then.

Reply:
done